src/test/scala/org/broadinstitute/dsde/workbench/sam/service/TestResourceServiceBuilder.scala

```diff
@@ -2,109 +2,31 @@ package org.broadinstitute.dsde.workbench.sam.service
 
 import cats.effect.IO
 import org.broadinstitute.dsde.workbench.openTelemetry.OpenTelemetryMetrics
-import org.broadinstitute.dsde.workbench.sam.api.TestSamRoutes.SamResourceActionPatterns
+import org.broadinstitute.dsde.workbench.sam.Generator.{genResourceType, genWorkspaceResourceType}
 import org.broadinstitute.dsde.workbench.sam.dataAccess.{AccessPolicyDAO, DirectoryDAO}
 import org.broadinstitute.dsde.workbench.sam.model._
 import org.mockito.scalatest.MockitoSugar
 
-import java.util.UUID
 import scala.collection.concurrent.TrieMap
 import scala.collection.mutable
 import scala.concurrent.ExecutionContext
 
 case class TestResourceServiceBuilder(
     policyEvaluatorService: PolicyEvaluatorService,
-    policyDAO: AccessPolicyDAO,
+    accessPolicyDAO: AccessPolicyDAO,
     directoryDAO: DirectoryDAO,
     cloudExtensions: CloudExtensions
 )(implicit val executionContext: ExecutionContext, val openTelemetry: OpenTelemetryMetrics[IO])
     extends MockitoSugar {
   private val emailDomain = "example.com"
-  private var maybeDummySamUser: Option[SamUser] = None
-  private var maybeDefaultResourceType: Option[ResourceType] = None
-  private var maybeOtherResourceType: Option[ResourceType] = None
-  private var maybeWorkspaceResourceType: Option[ResourceType] = None
-  private[service] val defaultResourceTypeActions =
-    Set(ResourceAction("alter_policies"), ResourceAction("delete"), ResourceAction("read_policies"), ResourceAction("view"), ResourceAction("non_owner_action"))
-  private[service] val defaultResourceTypeActionPatterns = Set(
-    SamResourceActionPatterns.alterPolicies,
-    SamResourceActionPatterns.delete,
-    SamResourceActionPatterns.readPolicies,
-    ResourceActionPattern("view", "", false),
-    ResourceActionPattern("non_owner_action", "", false)
-  )
+  private val resourceTypes: mutable.Map[ResourceTypeName, ResourceType] = new TrieMap()
 
-  def withRandomDefaultResourceType(): TestResourceServiceBuilder = {
-    maybeDefaultResourceType = Some(
-      ResourceType(
-        ResourceTypeName(UUID.randomUUID().toString),
-        defaultResourceTypeActionPatterns,
-        Set(
-          ResourceRole(ResourceRoleName("owner"), defaultResourceTypeActions - ResourceAction("non_owner_action")),
-          ResourceRole(ResourceRoleName("other"), Set(ResourceAction("view"), ResourceAction("non_owner_action")))
-        ),
-        ResourceRoleName("owner")
-      )
-    )
+  def withResourceTypes(): TestResourceServiceBuilder = {
+    Set(genResourceType.sample.get, genWorkspaceResourceType.sample.get)
+      .foreach(rt => resourceTypes += rt.name -> rt)
```

Comment:
I would prefer to see this construction of an arbitrary `Set` as part of the test.  Something like, 

```
"myTest" should "do the right thing" in {
    val resourceTypes = Set(genResourceType.sample.get, genWorkspaceResourceType.sample.get)
    val testResourceServiceBuilder = new TestResourceServiceBuilder()
        .withResourceTypes(resourceTypes)
        .build()
    … 
}
```
In this way, the builders are not responsible for creating any of the concrete objects that we care about in the test.  All of those concrete objects that the tests rely on are declared explicitly in the tests, making them easier to read and maintain.  

I guess you could put this another way and say it is a code smell if `org.broadinstitute.dsde.workbench.sam.Generator` is in your list of imports in one of these `Builder` classes.

Reply:
Refactored `withResourceTypes` to accept a collection of resource types as input.